### ISSUE TITLE
SimpleNamespace is not supported for implicit subtyping using Protocol

### ISSUE THREAD

**Issue by DerPhysikeR** [microsoft/pyright#4697]
**Describe the bug**
Pyright doesn't recognize it, if a Protocol is fulfilled by a SimpleNamespace instance, while mypy does.

**To Reproduce**

```
from types import SimpleNamespace
from typing import Protocol

class Chair(Protocol):
    def sit_on(self):
        ...

chair: Chair = SimpleNamespace(sit_on=lambda: "abc")
```

This results in the following error: `Expression of type "SimpleNamespace" cannot be assigned to declared type "Chair"    "SimpleNamespace" is incompatible with protocol "Chair"    "sit_on" is not present`

**Expected behavior**
No error message, like mypy, since `sit_on` is present.

**VS Code extension or command-line**
I run pyright v1.1.294 in neovim.

**Comment by DerPhysikeR** [comment]
After doing some more experiments I realized, that mypy also doesn't check if the SimpleNamespace fulfills the protocol, it just doesn't throw an error, which is actually explained in your comparison 

https://github.com/microsoft/pyright/blob/main/docs/mypy-comparison.md#type-checking-unannotated-code

Sorry, for the inconvenience.
Feel free to close this issue.

**Comment by erictraut** [comment]
If you're interested in static type checking, you should avoid using dynamic type creation mechanisms like `SimpleNamespace`. Instead, define the class using a `class` statement.

```python
class ChairProto(Protocol):
    def sit_on(self) -> str:
        ...

class Chair:
    def sit_on(self):
        return "abc"

chair: ChairProto = Chair()
```

**Comment by DerPhysikeR** [comment]
The idea was to keep the code which is for a demonstration of different design patterns succinct, by using SimpleNamespace in a few places instead of explicitly creating a lot of small classes, but it seems to be unavoidable if I want to use type annotations.
